meta-resin-common/classes/binary-compress.bbclass

```diff
@@ -0,0 +1,30 @@
+# e.g.  FILES_COMPRESS = "/path/bin1 /path/bin2 /path/bin3"
+# Define what binaries that we find in the package tree will be compress
+# This variable's definition is MANDATORY if this class is used
+# NOTE: path is relative to PKGD directory
+FILES_COMPRESS ?= ""
+
+DEPENDS_append = " upx"
+
+UPX ?= "${STAGING_BINDIR_NATIVE}/upx"
+UPX_ARGS ?= "--best -q"
+
+find_and_compress() {
+    #Sanity check
+    if [ -z ${FILES_COMPRESS} ]; then
+        bbfatal "Binary compress class imported but FILES_COMPRESS variable was found empty."
+    else
+        #Find all executables once
+        pkg_dir=${PKGD}/
+        execs=$(find $pkg_dir -executable -type f)
+        binaries=${FILES_COMPRESS}
+        #Compress
+        for exec in $execs; do
+            if echo $binaries | grep -q "${exec#$pkg_dir}" ; then
```

Comment:
This can fail if the package has `/usr/bin/a` and `/usr/bin/aa` and I set `FILES_COMPRESS` to be `"/usr/bin/a"`. Better check directly for the file using `if [ -x foobar ]`